src/overlays/actors/ovl_En_Raf/z_en_raf.c

```diff
@@ -80,65 +178,787 @@ static DamageTable D_80A1939C = {
     /* Powder Keg     */ DMG_ENTRY(0, 0xF),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80A18EE0;
-extern DamageTable D_80A1939C;
-
-extern UNK_TYPE D_06000108;
-extern UNK_TYPE D_060024E0;
-extern UNK_TYPE D_060032F8;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A16D40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A16D6C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/EnRaf_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/EnRaf_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17060.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A1712C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A171D8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17414.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17464.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17530.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A175E4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A17848.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A178A0.s")
+/**
+ * Sets the `index`th pixel of the trap teeth texture to 0 (transparent black)
+ * according to the `clearPixelTable`
+ */
+void EnRaf_ClearPixelsTeeth(u16* texture, u8* clearPixelTable, s32 index) {
+    if ((index < 0x40) && (clearPixelTable[index] != 0)) {
+        texture[index] = 0;
+    }
+}
+
+/**
+ * Sets the `index`th pixel of the trap petal texture to 0 (transparent black)
+ * according to the `clearPixelTable`
+ */
+void EnRaf_ClearPixelsPetal(u16* texture, u8* clearPixelTable, s32 index) {
+    if (clearPixelTable[index] != 0) {
+        texture[index] = 0;
+    }
+}
+
+void EnRaf_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnRaf* this = THIS;
+    Vec3f limbScale = { 1.0f, 1.0f, 1.0f };
+    s32 pad;
```

Comment:
Does the pad have to go exactly here?

Reply:
Yes, it has to be right there to get the stack to line up. I'm guessing the `GlobalContext *` temp was just there for some reason instead of at the top.

Reply:
Yeah, just unusual, that's all. Presumably the inline data forces the stack to be exactly this.

Reply:
Yeah, that's probably it. I remember having a devil of a time getting the stack in this function to work because of this pad.